compose/service/module.go

```diff
@@ -1223,12 +1219,49 @@ func moduleToModel(ctx context.Context, dmm dalModelManager, ns *types.Namespace
 			model.Attributes = append(model.Attributes, attrAux...)
 
 			out = append(out, model)
+
+			if mod.ModelConfig.Revisions.Enabled {
+				rModel := revisions.Model()
+
+				// reuse the connection from the module
+				rModel.ConnectionID = connectionID
+				rModel.Resource = model.Resource
+
+				if mod.ModelConfig.Revisions.DAL.Ident == "" {
+					mod.ModelConfig.Revisions.DAL.Ident = "compose_record_revisions"
+				}
+
+				rModel.Ident, err = makeModelIdent(ctx, ff, cm, mod, mod.ModelConfig.Revisions.DAL.Ident)
+				if err != nil {
+					return
+				}
+
+				out = append(out, rModel)
+			}
 		}
 	}
 
 	return
 }
 
+func makeModelIdent(ctx context.Context, ff identFormatter, cm dal.ConnectionMeta, mod *types.Module, ident string) (_ string, err error) {
+	var (
+		ok bool
+	)
+
+	if ident == "" {
+		ident = cm.DefaultPartitionFormat
+	}
+
+	ident, ok = ff.Format(ctx, ident, formatterModuleParams(mod)...)
+	if !ok {
+		err = fmt.Errorf("invalid model ident generated: %s", ident)
+		return
+	}
+
+	return ident, nil
+}
+
 // moduleFieldsToAttributes converts all user-defined module fields to attributes
 func moduleFieldsToAttributes(ctx context.Context, cm dal.ConnectionMeta, ns *types.Namespace, mod *types.Module) (out dal.AttributeSet, err error) {
```

Comment:
There is another fnc you'll need to update along with the struct change mentioned above (the added revision sys field comment) -- `partitionedModuleSystemFieldsToAttributes`